lib/bolt/cli.rb

```diff
@@ -446,7 +447,13 @@ def show_task(task_name)
 
     def list_tasks
       tasks = pal.list_tasks
-      tasks.select! { |task| task.first.include?(options[:filter]) } if options[:filter]
+      if options[:filter] || config.project.tasks.any?
+        tasks.select! do |task|
+          filter = options[:filter] ? task.first.include?(options[:filter]) : true
+          in_project = config.project.tasks.any? ? config.project.tasks.include?(task.first) : true
+          filter && in_project
+        end
+      end
```

Comment:
I think just adding a second `select!` keeps things a little more readable and avoids the need for the ternary operators.

```
tasks.select! { |task, _| config.project.tasks.include?(task) } if config.project.tasks.any?
```

Even with a list of thousands of tasks a second iteration goes quickly.

Reply:
I thought traversing the list twice would be a lot slower, but the time diff with ~600 tasks is only 0.5 seconds, which is pretty negligible.